js/tab.js

```diff
@@ -171,6 +180,32 @@ define([
     return syntaxValue;
   }
   
+  Tab.prototype.getWorker = function() {
+    return new Promise((resolve, reject) => {
+      // if worker is off, return null
+      if (!Settings.get("user").useWorker) return resolve(null);
+      
+      var checkLimit = 20;
+      var findWorker = () => {
+        // if worker was turned while we were waiting on it, return null
+        if (!Settings.get("user").useWorker) return resolve(null);
+      
+        // return the worker if we found it
+        if (this.$worker) {
+          return resolve(this.$worker);
+        }
+        
+        // see if we've checked too many times
+        if (checkLimit-- < 0) return reject(new Error('Could not find worker.'));
+        
+        // otherwise, go to sleep and then check again
+        setTimeout(findWorker, 100);
```

Comment:
I don't think you should need to do the timeout dance--createWorker() is called synchronously (https://github.com/ajaxorg/ace/blob/4a55188fdb0eee9e2d3854f175e67408a1e47655/lib/ace/edit_session.js#L955) and messages that are passed to workers are queued while it's initializing. I think you could probably replace this with a synchronous method and just return the worker or `null`.